Handle fileopen if running (second run)

HofiOne opened the pull request:
<!--- Provide a general summary of your change in the Title above -->

## Description
<!--- Describe your change in detail -->
Added the following features:
- accept media file open requests and replace the current playlist with a new one even if the app already started
- two new Finder service items, "Enqueue next in Kodi", "Enqueue last in Kodi" to modify the current playlists without replacing it
- ability to accept file drag and drop onto application icon
- listing Kodi in "Open with..." Finder context menu for common media file types

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Kodi Mac does not allow modifying or replacing the current playlist once the app has started.

<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Rechi commented:
@HofiOne please update your changes to fit our coding style.
Especially take a look at brackets (https://codedocs.xyz/xbmc/xbmc/code_guidelines.html#code_guidelines_2).

Rechi commented:
@HofiOne rebase on xbmc/master if you want the latest changes from it in your feature branch, but never ever merge master into a feature branch.

HofiOne commented:
@Rechi about the coding style, in most of the touched files (and others too i saw earlier) there are mixed styles already, even within the files and more often between different files, what style to follow this way?

Rechi commented:
It doesn't matter what styles are currently used in the files, the correct coding style is described at https://codedocs.xyz/xbmc/xbmc/code_guidelines.html#code_guidelines_2.
Only exception is, changing a few lines in an already existing function, then you can keep the original indentation to keep it readable.

Memphiz commented on xbmc/platform/darwin/osx/SDLMain.mm:
```diff
@@ -353,54 +374,142 @@ - (void) applicationDidFinishLaunching: (NSNotification *) note
+              NSURL* foundURL = [NSURL URLWithString:substringForMatch];
+              NSString* fileName = nil;
+              
+              if (foundURL.isFileReferenceURL || [foundURL.scheme caseInsensitiveCompare:@"file"] == NSOrderedSame)   // Already a file system path URL
```
This code is the same as in the else branch or am I blind?

Memphiz commented:
I would like to have this as at least 4 separated commits.

1. Info plist change for accepting the media 
2. Info plist change for added menu entries
3. that free after SDl Main
4. the whole enqueue features 

In case you spot some sane possibility to split the 4th commit even more this would be welcome aswell- but not mandatory.

Beside that - nice work man :)

HofiOne commented on xbmc/platform/darwin/osx/SDLMain.mm:
```diff
@@ -353,54 +374,142 @@ - (void) applicationDidFinishLaunching: (NSNotification *) note
+                fileName = foundURL.path;
+              else
+              {
+                foundURL = [NSURL fileURLWithPath:substringForMatch];
```
It's been a decade i've added this and at the first sight even i could not spot the difference
this is the difference, first trying to look for a match using URLWithString: second it uses fileURLWithPath:

HofiOne committed:
Handle fileopen if running

- Added support of opening media files even if Kodi already running that have been ignored earlier
- Instead of replacing always the current playlist in Kodi now we can add media files to playlist queue
- Add Enqueue next/last in Kodi services menu items
- Add media file handling support for "Open with ..." in Finder for common file types/extensions

phunkyfish commented:
PR closed as it appears to be abandoned by the author. 

If the author would like it kept open please just request so on the PR. If you feel there is still merit in merging this PR you can request the label be changed to `PR Cleanup: Backburner`.
